pkg/work/helper/helpers.go

```diff
@@ -364,20 +365,18 @@ func NewAppliedManifestWorkOwner(appliedWork *workapiv1.AppliedManifestWork) *me
 	}
 }
 
-func FindManifestConiguration(resourceMeta workapiv1.ManifestResourceMeta, manifestOptions []workapiv1.ManifestConfigOption) *workapiv1.ManifestConfigOption {
-	identifier := workapiv1.ResourceIdentifier{
-		Group:     resourceMeta.Group,
-		Resource:  resourceMeta.Resource,
-		Namespace: resourceMeta.Namespace,
-		Name:      resourceMeta.Name,
-	}
-
-	for _, config := range manifestOptions {
-		if config.ResourceIdentifier == identifier {
-			return &config
+func FindManifestConfiguration(resourceMeta workapiv1.ManifestResourceMeta,
+	manifestOptions []workapiv1.ManifestConfigOption) *workapiv1.ManifestConfigOption {
+	// return the first matched option
+	for i := 0; i < len(manifestOptions); i++ {
+		option := manifestOptions[i]
+		if resourceMeta.Group == option.ResourceIdentifier.Group &&
+			resourceMeta.Resource == option.ResourceIdentifier.Resource &&
+			wildcardMatch(resourceMeta.Namespace, option.ResourceIdentifier.Namespace) &&
+			wildcardMatch(resourceMeta.Name, option.ResourceIdentifier.Name) {
+			return &option
```

Comment:
1. If two options match, the first configured the FeedbackRules but no UpdateStrategy, and the second configured the UpdateStrategy but no FeedbackRules, will returning the first option cause the FeedbackRules to be lost? 
2. I am still considering if we should make the word-by-word matching take precedence over wildcard match?

Reply:
> If two options matched, the first configured the FeedbackRules but no UpdateStrategy, and the second configured the UpdateStrategy but no FeedbackRules, will returning the first option cause the FeedbackRules to be lost?

fixed by splitting the FindManifestConfiguration into 2 functions.

> I am still considering if we should make the word-by-word matching take precedence over wildcard matching?
I think the users only use the first option to filter resources in the most cases, so word-by-word matching is the same level as wildcard matching.
@qiujian16 WDYT?

Reply:
could we just build a new option finding the first available rule and the first updateStrategy in the FindManifestConfiguration?

Reply:
done. changed to one FindManifestConfiguration and return an option with the first matched updateStrategy and feedbackRule.